### PR TITLE
fixes display error for values below 0.01

### LINKED ISSUE #1738: Display <0.01 for small amounts
When inputting less than 0.01 for a transaction amount, the UI will display '0.00' in several places. Instead, it should display '<0.01'. Needs to be done in: buy, sell and pool views for both scalar and categorical markets.

Examples:
image
image

### PR DESCRIPTION
**Pull request by gem1no**
fixes #1738 
Changed formatNumber and logic for checking if value is below 0.01 to display correct information  WIP.

### CODE REVIEW

**Comment by gem1no** [comment]
It should be working on all markets now. Tested for categorial/scalar markets, and their respective buy/sell sides and pool deposit/withdrawals.

**Review by Mi-Lan — COMMENTED** [review]
Hey @gem1no seems to work in all places that is required! I just have couple of additions to make.
- Keeping DRY principle in mind i think you should have made function with that little piece of logic and imported it from tools. That would make it more reusable for the future use and also easier to modify

**Comment by Mi-Lan** [comment]
- Also i think it would be good to say 0 here initially instead of < 0.01
Screenshot 2021-03-22 at 12 49 19

File: app/src/components/market/sections/market_buy/market_buy.tsx [hunk 1]
```diff
@@ -251,7 +253,12 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
       )
       setDisplayAmountToFund(new BigNumber('0'))
       setStatus(Status.Ready)
-      setMessage(`Successfully bought ${sharesAmount} '${balances[outcomeIndex].outcomeName}' shares.`)
+      {
+        Number(sharesAmount) < 0.01
+          ? setMessage(`Successfully bought <${0.01} '${balances[outcomeIndex].outcomeName}' shares.`)
+          : setMessage(`Successfully bought ${sharesAmount} '${balances[outcomeIndex].outcomeName}' shares.`)
+      }
+
```
Comment (Mi-Lan) [inline app/src/components/market/sections/market_buy/market_buy.tsx:261]: I think we shouldn't repeat whole message and setMessage here but just make one function and number in it and make it format...

File: app/src/components/market/sections/market_buy/market_buy.tsx [hunk 2]
```diff
@@ -227,7 +227,9 @@ const MarketBuyWrapper: React.FC<Props> = (props: Props) => {
       const sharesAmount = formatBigNumber(displayTradedShares, baseCollateral.decimals)
 
       setStatus(Status.Loading)
-      setMessage(`Buying ${sharesAmount} shares ...`)
+      Number(sharesAmount) < 0.01
+        ? setMessage(`Buying <${0.01} shares`)
+        : setMessage(`Buying ${sharesAmount} shares ...`)
```
Comment (Mi-Lan) [inline app/src/components/market/sections/market_buy/market_buy.tsx:232]: same here

File: app/src/components/market/sections/market_buy/scalar_market_buy.tsx [hunk 3]
```diff
@@ -247,7 +247,9 @@ export const ScalarMarketBuy = (props: Props) => {
       const sharesAmount = formatBigNumber(tradedShares, collateral.decimals)
 
       setStatus(Status.Loading)
-      setMessage(`Buying ${sharesAmount} shares ...`)
+      Number(sharesAmount) < 0.01
+        ? setMessage(`Buying <${0.01} shares`)
+        : setMessage(`Buying ${sharesAmount} shares ...`)
```
Comment (Mi-Lan) [inline app/src/components/market/sections/market_buy/scalar_market_buy.tsx:252]: here as well

File: app/src/components/market/sections/market_buy/scalar_market_buy.tsx [hunk 4]
```diff
@@ -271,7 +273,12 @@ export const ScalarMarketBuy = (props: Props) => {
 
       setAmount(new BigNumber(0))
       setStatus(Status.Ready)
-      setMessage(`Successfully bought ${sharesAmount} '${balances[outcomeIndex].outcomeName}' shares.`)
+
+      {
+        Number(sharesAmount) < 0.01
+          ? setMessage(`Successfully bought <${0.01} '${balances[outcomeIndex].outcomeName}' shares.`)
+          : setMessage(`Successfully bought ${sharesAmount} '${balances[outcomeIndex].outcomeName}' shares.`)
+      }
```
Comment (Mi-Lan) [inline app/src/components/market/sections/market_buy/scalar_market_buy.tsx:281]: here

File: app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx [hunk 5]
```diff
@@ -286,7 +290,11 @@ const MarketPoolLiquidityWrapper: React.FC<Props> = (props: Props) => {
       setAmountToFundDisplay('')
       setAmountToFundNormalized(null)
       setStatus(Status.Ready)
-      setMessage(`Successfully deposited ${fundsAmount} ${displayCollateral.symbol}`)
+      {
+        Number(fundsAmount) < 0.01
+          ? setMessage(`Successfully deposited <${0.01} ${displayCollateral.symbol}`)
+          : setMessage(`Successfully deposited ${fundsAmount} ${displayCollateral.symbol}`)
+      }
```
Comment (Mi-Lan) [inline app/src/components/market/sections/market_pooling/market_pool_liquidity.tsx:297]: and all the rest :D

**Pull request closed without merging**